### ISSUE TITLE
build: Remove deprecated rules_docker from bazel build of container images

### ISSUE THREAD

**Issue by brianmcarey** [kubevirt/kubevirt#10983]
[rules_docker](https://github.com/bazelbuild/rules_docker) was deprecated in October 2023[1] 

KubeVirt uses rules_docker to handled the building and publishing of the kubevirt container images for example in virt-api[2].

It is recommended to move to [rules_oci](https://github.com/bazel-contrib/rules_oci) for handling the building and publishing of images.

[1] https://github.com/bazelbuild/rules_docker/commit/3040e1fd74659a52d1cdaff81359f57ee0e2bb41
[2] https://github.com/kubevirt/kubevirt/blob/99b6c4bdb1dcbc190f8e62408d70e3d0b14d00dc/cmd/virt-api/BUILD.bazel#L28C1-L52C2

**Comment by aburdenthehand** [comment]
@brianmcarey Is this a candidate for good first issue tag?

**Comment by brianmcarey** [comment]
> @brianmcarey Is this a candidate for good first issue tag?

I don't think so - I have a branch that I have tried to do this but hit issues. It could be a good candidate for `help-wanted`

**Comment by slashexx** [comment]
@brianmcarey Hi ! I'm thinking of working on this, are the changes only to be made in the file that you've linked ?

**Comment by brianmcarey** [comment]
> @brianmcarey Hi ! I'm thinking of working on this, are the changes only to be made in the file section that you've linked ? Or is it all the .build files under sub-cmd directories ?

Hi @slashexx - thanks for taking interest in this issue but I have already picked it and started working on it. I wouldn't consider it a good first issue as it is quite an important part of the kubevirt build system.

**Comment by slashexx** [comment]
@brianmcarey Oh sure, thanks for letting me know !

Also I apologize if this is not the correct place to ask this but can you suggest a good starting point if I want to start contributing here ? Can you please mention the prerequisites I should be proficient in ?

**Comment by brianmcarey** [comment]
@slashexx sorry for only getting back to you now, we generally tag any issues that are good for new contributors as good-first-issue so I think that would be a good place to start.
